stdlib/public/Concurrency/TaskCancellation.swift

```diff
@@ -0,0 +1,81 @@
+////===----------------------------------------------------------------------===//
+////
+//// This source file is part of the Swift.org open source project
+////
+//// Copyright (c) 2020 Apple Inc. and the Swift project authors
+//// Licensed under Apache License v2.0 with Runtime Library Exception
+////
+//// See https://swift.org/LICENSE.txt for license information
+//// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+////
+////===----------------------------------------------------------------------===//
+
+import Swift
+import Dispatch
+@_implementationOnly import _SwiftConcurrencyShims
+
+// ==== Task Cancellation ------------------------------------------------------
+
+extension Task {
+  public struct CancellationError: Error {
+    // no extra information
+  }
+
+  /// Returns `true` if the task is cancelled.
+  public static func isCancelled() async -> Bool {
+    fatalError("\(#function) not implemented yet.")
+  }
```

Comment:
We were not explicit about this so far, but I think this should also check for exceeding deadline if one was set on the task right? Otherwise functions will be forced to do a lot of checking (cancallation, deadline) before they start doing anything interesting.

We didn't formalize this yet, but was this the plan @rjmccall or are we not sure yet?